gammapy/scripts/cta_irf.py

```diff
@@ -392,24 +393,42 @@ def read(cls, filename, offset='0.5 deg'):
         ----------
         filename : `str`
             File containing the IRFs
+        e_reco : `~astropy.units.Quantity` array
+            Array of energies to reproject the background rate
         """
         filename = str(make_path(filename))
-
+        
         hdulist = fits.open(filename)
         aeff = EffectiveAreaTable.from_hdulist(hdulist=hdulist)
         edisp = EnergyDispersion2D.read(filename, hdu='ENERGY DISPERSION')
-        bkg = BgRateTable.from_hdulist(hdulist=hdulist)
         psf = Psf68Table.from_hdulist(hdulist=hdulist)
         sens = SensitivityTable.from_hdulist(hdulist=hdulist)
 
-        # Create rmf with appropriate dimensions (e_reco->bkg, e_true->area)
-        e_reco_min = bkg.energy.lo[0]
-        e_reco_max = bkg.energy.hi[-1]
-        e_reco_bin = bkg.energy.nbins
-        e_reco_axis = EnergyBounds.equal_log_spacing(
-            e_reco_min, e_reco_max, e_reco_bin, 'TeV',
-        )
+        # Reproject background if needed
+        e_reco_axis = None
+        bkg = None
+        if e_reco is None:
+            bkg = BgRateTable.from_hdulist(hdulist=hdulist)
```

Comment:
Both of your if / else start with the same line reading from file. Can you put this before the if / else?